### ISSUE TITLE
DottyDoc fails on valid overloaded constructor with js.UndefOr[_]

### ISSUE THREAD

**Issue by russwyte** [lampepfl/dotty#12215]
## Minimized example

```Scala
import scala.scalajs.js

class Foo(a: String, b: js.UndefOr[String]):
  def this(b: js.UndefOr[String]) = this("default", b)
```

## Output

```
[error] -- Error: src/main/scala/Main.scala:4:41 ---------------------------------------
[error] 4 |  def this(b: js.UndefOr[String]) = this("default", b)
[error]   |                                         ^^^^^^^^^^^^^
[error]   |undefined: this # -1: TermRef(ThisType(TypeRef(ThisType(TypeRef(ThisType(TypeRef(NoPrefix,module class <root>)),module class <empty>)),class Foo)),<init>) at readTasty
[warn] one warning found
[error] one error found
```

## Expectation
This code compiles correctly - so it seems DottyDoc should handle it correctly.

**Comment by russwyte** [comment]
@sjrd Interestingly - this similar code presents no issues with DottyDoc:

```Scala
import scala.scalajs.js

class Foo(a: String, b: Option[String]):
  def this(b: Option[String]) = this("default", b)
```

**Comment by griggt** [comment]
Looks like a duplicate of #11943

**Comment by russwyte** [comment]
This also seems to work fine - which seems odd to me because it's exactly the equivalent type of UndefOr :)

```Scala:
import scala.annotation.unchecked.uncheckedVariance
type UO[+A] = (A @uncheckedVariance) | Unit
class Foo(a: String, b: UO[String]):
  def this(b: UO[String]) = this("default", b)
```

**Comment by griggt** [comment]
Closing since this is fixed in Scala 3.0.1-RC1 / sbt 1.5.3
